src/ngraph/runtime/reference/generate_mask.hpp

```diff
@@ -37,6 +37,30 @@ namespace ngraph
                     out[i] = training ? static_cast<T>(bd(gen)) : static_cast<T>(1);
                 }
             }
+
+            template <typename T>
+            void generate_mask_no_state(
+                T* out, size_t count, bool training, uint32_t seed, double prob)
+            {
+                std::minstd_rand msr;
+                msr.seed(seed);
+                std::uniform_int_distribution<> dist(0, 1);
```

Comment:
shouldn't this be std::uniform_real_distribution? Alternatively, we could use bernoulli similar to the existing generate_mask

Reply:
In new commit, I changed similar to existing generate_mask without preserving state